addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -73,13 +73,18 @@ class ComposerTextInput extends Component {
         if (!this.composer) {
             return "";
         }
-        if (this.composer.thread && this.composer.thread.model !== 'mail.channel') {
-            if (this.composer.isLog) {
-                return this.env._t("Log an internal note...");
-            }
-            return this.env._t("Send a message to followers...");
+        if (!this.composer.thread) {
+            return "";
+        }
+        if (this.composer.thread.model === 'mail.channel') {
+            const name = this.composer.thread.displayName;
+            const placeholder = this.composer.thread.channel_type === 'chat'
+                              ? this.env._t("Message %s...")
+                              : this.env._t("Message #%s...");
+            return _.str.sprintf(placeholder, name);
```

Comment:
Here's what I had in mind:
```suggestion
        if (this.composer.thread.model === 'mail.channel') {
            if (this.composer.thread.correspondent) {
                return _.str.sprintf("Message %s...", this.composer.thread.correspondent.nameOrDisplayName)
            }
            return _.str.sprintf("Message #%s...", this.composer.thread.displayName);
        }
```
Also note the change about correspondent which also makes it work for livechat out of the box.